### PR TITLE
Upgrade to lit 2

### LINKED ISSUE #3196: Upgrade to Lit 2
### Description

The lit-html and lit-element dependencies are pretty out of date at this point. Upgrading to Lit 2 should be relatively easy for this project.

### Version
- model-viewer: current

### PR DESCRIPTION
**Pull request by e111077**
I have upgraded model-viewer and space-opera to Lit 2. You can see the changes required in the upgrade guide:

https://lit.dev/docs/releases/upgrade/

I had trouble getting the tests to run locally. I was not sure if it's something up with my local build or maybe CI could get it. Running tests manually in browsers passed except for some Safari failures in space-opera?

Specific changes I've made:

* Upgrade all local imports of lit-element & lit-html to Lit 2
* `_getUpdateComplete` -> `getUpdateComplete`
* `await this.requestUpdate()` -> `this.requestUpdate();await this.updateComplete;`
* `UpdatingElement` -> `ReactiveElement`
* `@internalProperty` -> `@state`
* upgraded to the version of MWC that supports Lit 2
  * problem with MWC .d.ts types means I had to add `skipLibCheck` to space opera tsconfig
* implemented own spread directive
  * added tests for spread directive
* removed `@open-wc/lit-helpers` since it relied on Lit 1 and we were no longer using that spread directive
* Fixed some lit-plugin lint issues e.g. `@property({type: Function})` -> `@property({attribute: false})`

### Reference Issue

Fixes #3196

### CODE REVIEW

File: packages/space-opera/src/components/utils/render_model_viewer.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ export function renderModelViewer(
     childElements?: Array<TemplateResult|HTMLElement>) {
   const skyboxImage =
       config.useEnvAsSkybox ? config.environmentImage : undefined;
-  return html`<model-viewer ...=${spread(extraAttributes)}
+  return html`<model-viewer ${spread(extraAttributes)}
```
Comment (elalish) [inline packages/space-opera/src/components/utils/render_model_viewer.ts:98]: Thanks! This was the part I was struggling with. I've done some testing locally and it all seems to work great, but I noticed one odd thing that I can't seem to find the source of: 
image
Specifically this: `<!--?lit$515089429$-->`
Not that it's really hurting anything, but it's ugly and slightly ominous...

**Review by elalish — CHANGES_REQUESTED** [review]
Don't worry about our automated tests; we're having some trouble with BrowserStack at the moment that's not your fault. I really appreciate the help here!

File: packages/space-opera/src/components/utils/render_model_viewer.ts (hunk 1 above)
Comment (e111077) [reply packages/space-opera/src/components/utils/render_model_viewer.ts:98]: Ah, that's new in Lit 2, used for keeping track of Lit template binding locations. I can see if I can scratch something up to try and strip them from that textarea if you'd like?

Comment (elalish) [reply packages/space-opera/src/components/utils/render_model_viewer.ts:98]: Yes, that'd be very helpful. This is user-facing, so I'd prefer not to confuse them with internal details.

Comment (e111077) [reply packages/space-opera/src/components/utils/render_model_viewer.ts:98]: done! see:

[`0de10eb` (#3405)](https://github.com/google/model-viewer/pull/3405/commits/0de10eb0a3fb7f93999beb8f7a5cab824b7b6897)

**Review by elalish — APPROVED** [review]
Fantastic, thank you!

**Comment by e111077** [comment]
Happy to help! I know model-viewer is high traffic. Don't hesitate to @ me or ping me internally if you need any support regarding Lit 2

**Pull request merged**